dev/table.h

```diff
@@ -26,28 +26,31 @@ namespace sqlite_orm {
              */
             std::string name;
 
-            bool _without_rowid = false;
+            const bool _without_rowid;
         };
 
+        template<class T, class... Cs>
+        struct table_without_rowid_t;
+
         /**
-         *  Table interface class. Implementation is hidden in `table_impl` class.
+         *  Template for table interface class. Implementation is hidden in `table_impl` class.
          */
-        template<class T, class... Cs>
-        struct table_t : table_base {
+        template<class T, bool WR, class... Cs>
+        struct table_template : private table_base {
```

Comment:
why `private` inheritance? Is there any reason for this?

Reply:
Private inheritance prevents conversions from 'table_t ' to 'table_without_rowid_t' (and vice versa).
Perhaps this is excessive paranoia ..
Maybe we dont need 'table_base' structure? Let's just move the 'name' parameter to 'table_template'?